src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -22,6 +22,16 @@ def _resolve_aliases(target):
     else:
       yield declared
 
+    for export in _resolve_exports(declared):
+      yield export
+
+
+def _resolve_exports(target):
+  for export in getattr(target, 'exports', []):
+    for exp in _resolve_exports(export):
```

Comment:
Just to clarify between the two scenarios:
1. If A exports B, B exports C, then C should be available for A at compile time, which is what this bazel discussion is about https://groups.google.com/d/msg/bazel-discuss/CRd1BbkHVUw/8Vi1gPAjIAAJ
2. If A depends on B, B exports C, C exports D, is D available for A at compile time? (I am not too clear about this one)